File: cgi-bin/image.php
#!/usr/bin/php
<?php
$image = file_get_contents("./html/images/favicon_96.png");
echo("Accept-Ranges: bytes\n");
echo("Content-Length: 7936\n");
echo("Content-Type: image/png\n");
echo("Connection: keep-alive\n\n");
echo($image);
?>

File: cgi-bin/image.pl
#!/usr/bin/perl -w
my $file = "./html/images/42.png";
my $length = -s $file;
print "Accept-Ranges: bytes\n";
print "Content-Length: $length \n";
print "Content-Type: image/png\n";
print "Connection: keep-alive\n\n";

binmode STDOUT;
open (FH,'<', $file) || die "Could not open $file: $!";
my $buffer = "";
while (read(FH, $buffer, 1024)) {
    print $buffer;
}